File: solutions/3127. Make a Square with the Same Color/3127.py
# Time complexity: O(1)
# Space complexity: O(1)
class Solution:
  def canMakeSquare(self, grid: List[List[str]]) -> bool:
    for i in range(len(grid) - 1):
      for j in range(len(grid[0]) - 1):
        b, w = 0, 0
        for u, v in [(i, j), (i + 1, j), (i, j + 1), (i + 1, j + 1)]:
          b += int(grid[u][v] == 'B')
          w += int(grid[u][v] == 'W')
        if b >= 3 or w >= 3:
          return True
        
      return False

File: solutions/3127. Make a Square with the Same Color/3127.cpp
// Time complexity: O(1)
// Space complexity: O(1)
class Solution {
 public:
  bool canMakeSquare(const vector<vector<char>>& grid) {
    for (int i = 0; i < grid.size() -  1; ++i)
      for (int j = 0; j < grid[0].size() - 1; ++j) {
        int b = 0, w = 0;
        for (auto& p : vector<pair<int, int>>{{i, j}, {i + 1, j},
                                               {i, j + 1}, {i + 1, j + 1}}) {
          int u = p.first;
          int v = p.second;
          b += int(grid[u][v] == 'B');
          w += int(grid[u][v] == 'W');
        }
        if (b >= 3 || w >= 3)
          return true;
      }

    return false;
  }
};